### ISSUE TITLE
Requests are not being sent to PiHole when set via router settings

### ISSUE THREAD

**Issue by TheOneWho** [pi-hole/pi-hole#458]
##### Actual Behavior: Google DNS is handling requests

When I set my DNS servers in Windows PiHole works great, but when set through my router's settings it does not use PiHole.

My Router is a WNDR3700 by Netgear

[Router Settings](http://i.imgur.com/r76daPl.png)

[Example](http://i.imgur.com/FlusNSG.png)

[Ipconfig on the Windows machine](http://i.imgur.com/mjoP65I.png)

**Comment by PromoFaux** [comment]
Hmm, I'm not familiar with that router, but that looks about right. What happens if you manually set your windows machine to have the pi as DNS?

**Comment by TheOneWho** [comment]
When I set Windows to use the Pi as DNS it works perfectly

**Comment by PromoFaux** [comment]
Hmm, OK that rules out the pi not working!

Looks like this could be down to your router. Are there any other settings for DNS in the menu?

**Comment by PromoFaux** [comment]
Also worth nothing that once you've changed the DNS settings on your router, you'll need to reboot your windows PC (Or renew it's DHCP settings by running `ipconfig /renew`)

But from your screenshot, it looks like [you've set it up right](http://m.setuprouter.com/router/netgear/wndr3700/dns.htm)

**Comment by TheOneWho** [comment]
I have restarted my computer since changing the dns settings in the router. I have not managed to find any other DNS related options on the routers config.

**Comment by PromoFaux** [comment]
Could you paste a full screenshot of that router settings page, please?

**Comment by TheOneWho** [comment]
[Sure!](http://i.imgur.com/50cg6YI.png)

**Comment by PromoFaux** [comment]
OK, and under the "Advanced" section, a screenshot of `WAN setup` and `LAN setup` pages, please.

**Comment by TheOneWho** [comment]
[WAN](http://i.imgur.com/7uPUwZy.png)
[LAN](http://i.imgur.com/DssZBFO.png)

**Comment by PromoFaux** [comment]
ha, nothing of interest there.

I wonder, are there any settings anywhere for disabling "DNS Rebind protection"? We've seen issues before with some routers that don't allow routing DNS to a local address. Some have a setting you can switch off to enable that, some don't! You may have better luck flashing the router with `dd-wrt`, depending on your confidence level when it comes to flashing alternative firmwares...

**Comment by TheOneWho** [comment]
I was actually just looking at DD-WRT. Would flashing my routers firmware with dd-wrt have a good chance of helping with this issue? I couldn't find anything that looked like Rebind protection in the settings.

**Comment by jacobsalmela** [comment]
Everyone I have talked to that is running DD-WRT has been able to get it to work.  That's no guarantee of course, but you'll probably have better luck than you are now.

**Comment by TheOneWho** [comment]
I have flashed DD-WRT to my router and everything works fine now.

**Comment by gawdsed** [comment]
Silly question, but after you set your DHCP to your local IP address, did you re-obtain your DHCP address?

ipconfig -release
ipconfig -renew

then ipconfig /all to see your DNS configurations.

You also shouldn't need to put google's DNS address in there, you should be able to leave it blank. 

Another thing, did you reboot the router after? some of them are picky like that.

Lastly, is the firmware updated on that router?

**Comment by TheOneWho** [comment]
I restarted the router a few times, and restarted my computer a few times. I did not try to re-obtain my DHCP address but I connected a device to the network that hadn't ever been connect to this particular router and pihole did not receive and dns requests. The firmware on the router was the latest version I could find on the internet. I flashed DD-WRT onto the router today and now the Pi is properly relieving dns requests and works perfectly now.
